frontend/src/reducers/inbox/index.js

```diff
@@ -77,6 +77,42 @@ export function markThreadAsRead(threadId) {
   };
 }
 
+export const emailOpen = dataToPost => async dispatch => {
+  const {
+    excerciseAction,
+    associationsDetails,
+    time,
+    excerciseID,
+  } = dataToPost;
+
+  const url = `exercise/${excerciseID}/action/`;
+  const rawResponse = () =>
+    fetch(url, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        milliseconds: time,
+        action: {
+          type: excerciseAction,
+          ...associationsDetails,
+        },
+      }),
+    });
+
+  try {
+    const resJSON = await rawResponse();
```

Comment:
You need to call `.json()` to actually get the json response

Reply:
lol yup...classic poor choice of variable name. :D